control_protocol.md

```diff
@@ -0,0 +1,232 @@
+:::{mermaid}
+graph TD
+    CA([CA]) -->|"Request A"| D[recv] -->|"IA|Request A"| A([Coordinator])
+    CB([CB]) -->|"Request B"| C[recv] -->|"IB|Request B"| A
+    subgraph ROUTER socket
+        B
+        C
+        D
+        E
+    end
+    A -. "IA|Reply A" .-> B
+    B[send] -. "Reply A" .-> CA
+    A -. "IB|Reply B" .-> E
+    E[send] -. "Reply B" .-> CB
+:::
+
+
+### Naming scheme
+
+Each Component has an individual ID, given by the user, the Component ID.
+A Component ID must be a series of bytes, without the ASCII character "." (byte value 46).
+Component IDs must be unique in a Node, i.e. among the Componets connected to a single Coordinator.
+
+As each Component belongs to exactly one Coordinator, it is fully identified by the combination of Node ID and Component ID.
+This _full ID_ is the composition of Node ID, ".", and Component ID.
+
+The receiver of a message may be specified without the Node ID, if the receiver lives in the same Node.
+
+:::{note}
+How to address the Coordinator? I used "no Recipient".
+Or a fixed name?
+:::
+
+
+### Message composition
+
+A message consists in two or more frames.
+1. The receiver full ID.
+2. The sender full ID.
+3. Message content: The optional payload, which can be 0 or more frames.
+
+
+### Conversation protocol
+
+In the protocol examples, `CA`, `CB`, etc. indicate Components.
+`Co1`, `Co2`, etc. indicate Coordinators.
+
+Here the Message content is expressed in plain English, for the exact definition see {ref}`message-layer`.
+
+In the exchange of messages, only the messages over the wire are shown, the connection identity of the ROUTER socket is not shown.
+
+
+#### Communication with the Coordinator
+
+##### Initial connection
+
+After connecting to a Coordinator (`Co1`), a Component (`CA`) shall send a CONNECT message indicating its ID.
+The Coordinator shall respond with an ACKNOWLEDGE, giving the Node ID and other relevant information, or with an ERROR, if the ID is already taken.
+After that successful handshake, the Coordinator shall store the connection identity and correspondind Component ID in its address book.
+:::{note}
+publish that information
+:::
+Similarly, the Component shall store the Node ID and use it from this moment to generate its full ID.
+
+:::{mermaid}
+sequenceDiagram
+    Note over CA,Co1: ID "CA" is still free
+    CA ->> Co1: ||CA|CONNECT
+    Note right of Co1: Connection identity "IA"
+    Note right of Co1: Stores "CA" with identity "IA"
+    Co1 ->> CA: Co1.CA||ACKNOWLEDGE: Node ID is "Co1"
+    Note left of CA: Stores "Co1" as Node ID
+    Note over CA,Co1: ID "CA" is already used
+    CA ->> Co1: ||CA|CONNECT
+    Co1 ->> CA: CA||ERROR: ID "CA" is already used.
+    Note left of CA: May retry with another ID
+:::
+
+
+##### Heartbeat
+
+We use heartbeat to know, whether a communication partner is still online.
+
+Every message received counts as a heartbeat.
+
+:::{note}
+TBD: Respond to every non empty message with an empty one?
```

Comment:
Sounds good to me, pending discussion in #40

Reply:
I lament the loss of symmetry in messages, but if there's no other way to track message receipt in zmq, I guess it cannot be avoided. Deeper discussion in #40.